docs/java-helloworld.md

```diff
@@ -0,0 +1,145 @@
+
+## Project setup
+
+Open IntelliJ and create a new Gradle project by following Step 1 of the [Getting started with Gradle guide](https://www.jetbrains.com/help/idea/getting-started-with-gradle.html#create_project). Make sure your project is created inside the "temporal-java-tutorials" directory and name your project "helloworld-tutorial". It will take a few moments to complete.
+
+Once Gradle has finished scaffolding you will need to customize the project dependencies. To do this, open the build.gradle file that is in the root of your project. Add the following lines to the dependencies section while replacing {version} with the most recent version listed on the [Temporal SDK Maven repository page](https://search.maven.org/artifact/io.temporal/temporal-sdk):
+
+```
+implementation 'io.temporal:temporal-sdk:{version}'
+implementation group: 'ch.qos.logback', name: 'logback-classic', version: '1.2.3'
+```
+
+The 'logback-classic' dependency will ensure that there is a logger to bind to within the SDK and prevent a default logger warning message.
+
+A "refresh" icon will appear on the screen, click it to load the changes. Gradle will rebuild with the dependencies.
+
+Lastly, to limit the logging output from the SDK, within src/main/resources create a logback.xml file and paste in the following XML:
+
+<!--SNIPSTART java-samples-logback-dependency-configuration-->
+<!--SNIPEND-->
+
+## Build "Hello World!" app
+
+Now we are ready to create our Temporal Workflow application.
+
+Within src/main/java/ create the "helloworld" directory and within src/main/java/helloworld create the following files:
+
+- HelloWorldActivity.java
+- HelloWorldActivityImpl.java
+- HelloWorldWorkflow.java
+- HelloWorldWorkflowImpl.java
+- HelloWorldStarter.java
+- HelloWorldWorker.java
+
+The following code snippets can be copied and pasted directly into your Java files. However, make sure that the first line of each file specifies the package name:
+
+```java
+package helloworld;
+```
+
+### Activity
+
+First, let's define our Activity. Activities are meant to handle non-deterministic code that could result in unexpected results or errors. But for this tutorial all we are doing is taking a string, appending it to "Hello", and returning it back to the Workflow.
+
+Open HelloWorldActivity.java and define the Activity interface:
+
+<!--SNIPSTART java-hello-world-sample-activity-interface-->
+<!--SNIPEND-->
+
+### Workflow
+
+A Workflow is a special function that organizes the sequence of Activity executions. Opposed to an Activity, a Workflow is deterministic in nature. It is where you can specify Activity options, such as what to do if the Activity results in an error or when to time out the Activity execution. Since our Activity is just formatting strings we will only specify the required timeout.
+
+Open HelloWorldWorkflow.java and define the Workflow interface:
+
+<!--SNIPSTART java-hello-world-sample-workflow-interface-->
+<!--SNIPEND-->
```

Comment:
We need to explain what activity stub is. It is the most tricky part of the code.

Reply:
I completely agree. I almost think the best route is by making it seem very complex. Why do I need this verbose stub? Well it enables you to make automatically loadbalanced, stateful RPC calls with a native function invocation. A stub is a small price to pay for such functionality.

Reply:
Also I should comment on the samples PR but I'm lazy. I think the comments in the Workflow impl are sort of not helpful. "Set the ActivityOptions" tells me nothing and definitely could be autogenerated comment. I think comments are great but not if they replace reading the code.

Reply:
This isn't perfect but it's bette rthan the other comments

"    // Execute the Activity within the Workflow method
    // If there were more Activities, they would be executed from within this method"